File: musheng-tax-web/src/types/shop.ts
/**
 * 店铺管理类型定义
 * 与后端 ShopController 接口对齐
 */

// 店铺信息（对应后端 Shop 实体）
export interface Shop {
  id: number
  shopCode: string        // 店铺编码
  shopName: string        // 店铺名称
  sellerId?: string       // 亚马逊卖家ID
  companyName?: string    // 公司名称
  taxId?: string          // 统一社会信用代码
  status: number          // 状态 1-启用 0-禁用
  remark?: string         // 备注
  createTime: string
  updateTime?: string
}

// 店铺查询参数（对应后端 ShopQueryRequest）
export interface ShopQuery {
  shopCode?: string       // 店铺编码（模糊匹配）
  shopName?: string       // 店铺名称（模糊匹配）
  status?: number         // 状态
  page?: number
  size?: number
}

// 店铺创建/更新参数（对应后端 ShopRequest）
export interface ShopForm {
  id?: number
  shopCode: string
  shopName: string
  sellerId?: string
  companyName?: string
  taxId?: string
  status?: number
  remark?: string
}

File: musheng-tax-system/musheng-config/src/main/java/com/musheng/config/shop/entity/Shop.java
package com.musheng.config.shop.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.musheng.common.entity.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 店铺实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_shop")
@Schema(description = "店铺实体")
public class Shop extends BaseEntity {

    /**
     * 店铺编码
     */
    @Schema(description = "店铺编码")
    private String shopCode;

    /**
     * 店铺名称
     */
    @Schema(description = "店铺名称")
    private String shopName;

    /**
     * 亚马逊卖家ID
     */
    @Schema(description = "亚马逊卖家ID")
    private String sellerId;

    /**
     * 公司名称
     */
    @Schema(description = "公司名称")
    private String companyName;

    /**
     * 统一社会信用代码
     */
    @Schema(description = "统一社会信用代码")
    private String taxId;

    /**
     * 状态(1启用/0禁用)
     */
    @Schema(description = "状态(1启用/0禁用)")
    private Integer status;

    /**
     * 备注
     */
    @Schema(description = "备注")
    private String remark;
}
